Exclude compiled and minified JS and CSS assets from Git

kloon opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:
As discussed on numerous occasions and recently amongst the team, this PR removes compiled and minified CSS and JS assets from Git.

The reasoning is to mainly eliminate merge conflicts, but also make it easier to contribute JS and CSS without too much confusion.

When developing and making CSS or JS changes one always needs to have SCRIPT_DEBUG set to true as well as run `grunt watch` this will ensure that changes are shown when made.

When committing changes to CSS only the SCSS file changes must be committed, for JS only changes to the .js file must be committed and not the .min.js file.

Our deploy script has been updated to include building these assets and committing to SVN when deploying to wp.org. The only other process that may need attention is releases on GitHub, but if I recall that is also handled via our release script so should be fine cc @claudiosanches 

Would love some additional feedback from @woocommerce/proton before merging this to see if there are any other angles that need to be covered. We might need to update our e2e tests as well to compile and minify assets before running.

### TODO before merging:
* [x] Update contributing wiki to describe how to contribute CSS/JS changes
* [x] Update contributing wiki to reflect how to test CSS/JS changes or why CSS/JS files might be missing when cloning from GitHub
* [ ] Update FG with details on the CSS/JS changes. 

### How to test the changes in this Pull Request:

1. Checkout this branch and ensure that there are no *.min.js files in assets/js/admin and assets/js/frontend. Also check assets/css folder does not contain any *.css files.
2. Now run `grunt js` and `grunt css` this should compile and minify all the css and js files
3. Try a git commit, no files should be available for commit.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

kloon committed:
Exclude compiles CSS and minified JS files from repository.

kloon committed:
Remove compiled and minified CSS from git

kloon committed:
Revert "Remove compiled and minified CSS from git"

This reverts commit 1850d16d9fe94bd3faa39dca5375900e0f79a903.

kloon committed:
Remove compiled and minified css from git, except select2.css as that is a library

kloon committed:
Remove all admin minified js files

kloon committed:
Remove all frontend minified js files

kloon committed:
Include select2.css in ignore as it is generated via grunt as well.

kloon committed:
Run grunt css and js before running e2e tests

claudiosanches commented:
> The only other process that may need attention is releases on GitHub, but if I recall that is also handled via our release script so should be fine

For creating the release on GitHub should be just fine, but this make impossible to properly generate the .zip file that we usually create using `git archive`, but we can make the release script already generate that file for us.

Also I like to suggest some shortcut for `grunt js && grunt css`, since `grunt dev` copy blocks code and do more things, or even move blocks from the default/dev task.

Another thing that should be nice to have it's something to detected if we have the compressed files, and if not, display a warning in the admin, the same as woocommerce-admin is already doing:
https://github.com/woocommerce/woocommerce-admin/blob/d00204798475a5c1da5151822346a452ec649c40/woocommerce-admin.php#L96-L107

kloon committed:
Add notice when assets are not built, introduce grunt assets task for compiling and minifying assets.

kloon committed:
Use plugin_path instead of plugin_url.

kloon commented:
Added check in wp-admin with notice and also introduced a new grunt assets task for just compiling and minifying assets.

![](https://cld.wthms.co/QmO0xR+)

claudiosanches reviewed (approved):
Great work!

kloon committed:
Merge branch 'master' into update/exclude-compiled-assets-from-git
